plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergModule.java

```diff
@@ -130,10 +130,18 @@ public void configure(Binder binder)
         newOptionalBinder(binder, IcebergFileSystemFactory.class).setDefault().to(DefaultIcebergFileSystemFactory.class).in(Scopes.SINGLETON);
         newOptionalBinder(binder, CacheKeyProvider.class).setBinding().to(IcebergCacheKeyProvider.class).in(Scopes.SINGLETON);
 
+        closingBinder(binder).registerExecutor(ExecutorService.class);
         closingBinder(binder).registerExecutor(Key.get(ListeningExecutorService.class, ForIcebergSplitManager.class));
         closingBinder(binder).registerExecutor(Key.get(ExecutorService.class, ForIcebergScanPlanning.class));
     }
 
+    @Singleton
+    @Provides
```

Comment:
I would keep existing convention of having an annotation (@ForIcebergMetadata) even if it's not strictly needed